default/scripting/buildings/buildings.py

```diff
@@ -132,3 +138,29 @@
         ],
     ),
 ]
+
+
+CAN_ADD_STARLANE_TO_SOURCE = (
+    System
+    & ~Contains(IsSource)
+    & ~HasStarlane(from_=Object(id=Source.SystemID))
+    # specify the system, not the source object. the condition will consider multiple lanes
+    # connecting to a system as not crossing,
+    # but if the object is not itself a system but is in a system,
+    # then the system may be considered as crossing a lane from the object's location
+    & ~StarlaneToWouldCrossExistingStarlane(condition=Object(id=Source.SystemID))
```

Comment:
It accepts `from_` argument have like in `HasStarlane` or other places where were `from` in old FOCS